ee/tabby-webserver/src/service/answer.rs

```diff
@@ -489,6 +500,91 @@ Remember, don't blindly repeat the contexts verbatim. When possible, give code s
     )
 }
 
+/// Combine code snippets from search results rather than utilizing multiple hits: Presently, there is only one rule: if the number of lines of code (LoC) is less than 200, and there are multiple hits (number of hits > 1), include the entire file.
+pub async fn merge_code_snippets(
+    repository: Arc<dyn RepositoryService>,
+    hits: Vec<CodeSearchHit>,
+    policy: &AccessPolicy,
+    source_id: &str,
+) -> Vec<CodeSearchHit> {
+    // group hits by filepath
+    let mut file_hits: HashMap<String, Vec<CodeSearchHit>> = HashMap::new();
+    for hit in hits.into_iter() {
+        file_hits
+            .entry(hit.doc.filepath.clone())
+            .or_default()
+            .push(hit);
+    }
+
+    let mut result = Vec::with_capacity(file_hits.len());
+
+    for (filepath, file_hits) in file_hits {
+        if file_hits.len() > 1 {
+            // more than 1 hit in the same file, attempt to merge
+            let repo = match repository
+                .resolve_repository_by_source_id(policy, source_id)
+                .await
```

Comment:
instead of invoking this repeatly, should:
1. call repository_list before line 521 and store it in variable X
2. find repository by source id from X.